apps/sms/services/test/unit/conversation/conversation_service_test.js

```diff
@@ -0,0 +1,318 @@
+/*global bridge,
+         BroadcastChannel,
+         ConversationService,
+         Drafts,
+         MocksHelper
+*/
+
+'use strict';
+
+require('/services/test/unit/mock_bridge.js');
+require('/services/test/unit/mock_drafts.js');
+require('/services/js/bridge_service_mixin.js');
+require('/views/shared/test/unit/mock_broadcast_channel.js');
+require('/services/js/conversation/conversation_service.js');
+
+var MocksHelperForAttachment = new MocksHelper([
+  'bridge',
+  'BroadcastChannel',
+  'Drafts',
+  'Draft',
+  'streamClient',
+  'streamService'
+]).init();
+
+suite('ConversationService >', function() {
+  var serviceStub, mobileMessageClientStub, mobileMessageStreamStub;
+
+  MocksHelperForAttachment.attachTestHelpers();
+
+  setup(function() {
+    serviceStub = sinon.stub({
+      method: () => {},
+      stream: () => {},
+      broadcast: () => {},
+      listen: () => {},
+      plugin: () => {}
+    });
+
+    mobileMessageClientStub = sinon.stub({
+      stream: () => {},
+      plugin: () => {},
+      disconnect: () => {}
+    });
+    mobileMessageClientStub.plugin.returns(mobileMessageClientStub);
+
+    this.sinon.spy(self, 'BroadcastChannel');
+
+    this.sinon.stub(bridge, 'service').withArgs('conversation-service').returns(
+      serviceStub
+    );
+
+    this.sinon.stub(bridge, 'client').withArgs({
+      service: 'mozMobileMessageShim',
+      endpoint: sinon.match.instanceOf(BroadcastChannel).and(
+        sinon.match.has('name', 'shim-channel-1')
+      )
```

Comment:
> do you really need to be that specific ?

Yep, I think it's better to be on the safe side with all these channels and etc.

> could even be the full args: 

Looks good, thanks!